triggers/astarte_trigger_test.go

```diff
@@ -0,0 +1,707 @@
+		"action":"",
+		"simple_triggers": [
+		  {
+			"type": "invalid_trigger",
+			"on": "incoming_data",
+			"interface_name": "*",
+			"match_path":"/*",
+			"value_match_operator":"==",
+			"known_value":"3"
+		  }
+		]
+	  }`
+
+	_, err = ParseTriggerFrom([]byte(DeviceTriggerNoAction))
+	if err == nil {
+		t.Error("This trigger should have Failed! required action")
+
+	} else {
+		fmt.Println(err)
+	}
+
+	DeviceTriggerNoTriggers := `
+	{
+		"name": "test",
+		"action": {
+			"http_url": "https://example.com/my_hook",
+			"http_method": "post"
+		},
+		"simple_triggers": [
+		
+		]
+	  }`
+
+	_, err = ParseTriggerFrom([]byte(DeviceTriggerNoTriggers))
+	if err == nil {
+		t.Error("This trigger should have Failed! invalid json match operator")
+
+	} else {
+		fmt.Println(err)
+	}
+	DeviceTriggerTooManyTriggers := `
+	{
+		"name": "test",
+		"action": {
+			"http_url": "https://example.com/my_hook",
+			"http_method": "post"
+		},
+		"simple_triggers": [
+		  {
+			"type": "device_trigger",
+			"on": "incoming_data",
+			"interface_name": "*",
+			"match_path":"/*",
+			"value_match_operator":"==",
+			"known_value":"3"
+		  },
+		  {
+			"type": "device_trigger",
+			"on": "incoming_data",
+			"interface_name": "*",
+			"match_path":"/*",
+			"value_match_operator":"==",
+			"known_value":"3"
+		  }
+		]
+	  }`
+
+	_, err = ParseTriggerFrom([]byte(DeviceTriggerTooManyTriggers))
+	if err == nil {
+		t.Error("This trigger should have Failed! invalid json match operator")
+
+	} else {
+		fmt.Println(err)
+	}
+	DeviceTriggerTypeNotSet := `
+	{
+		"name": "test",
+		"action": {
+			"http_url": "https://example.com/my_hook",
+			"http_method": "post"
+		},
+		"simple_triggers": [
+		  {
+			"on": "incoming_data",
+			"interface_name": "*",
+			"match_path":"/*",
+			"value_match_operator":"==",
+			"known_value":"3"
+		  }
+		]
+	  }`
+
+	_, err = ParseTriggerFrom([]byte(DeviceTriggerTypeNotSet))
+	if err == nil {
+		t.Error("This trigger should have Failed! invalid json match operator")
+
+	} else {
+		fmt.Println(err)
+	}
+
+	DeviceTriggerNoDeviceORGroup := `
+	{
+		"name": "test",
+		"action": {
+			"http_url": "https://example.com/my_hook",
+			"http_method": "post"
+		},
+		"simple_triggers": [
+		  {
+			"type": "device_trigger",
+			"on": "device_connected"
+		  }
+		]
+	  }`
+
+	_, err = ParseTriggerFrom([]byte(DeviceTriggerNoDeviceORGroup))
+	if err == nil {
+		t.Error("This trigger should have Failed! invalid json match operator")
+
+	} else {
+		fmt.Println(err)
+	}
+	DeviceTriggerInvalidData := `
+	{
+		"name": "test",
+		"action": {
+			"http_url": "https://example.com/my_hook",
+			"http_method": "post"
+		},
+		"simple_triggers": [
+		  {
+			"type": "device_trigger",
+			"on": "device_connected",
+			"device_id": "dsagfsda",
+			"interface_name": "*"
+
+		  }
+		]
+	  }`
+
+	_, err = ParseTriggerFrom([]byte(DeviceTriggerInvalidData))
+	if err == nil {
+		t.Error("This trigger should have Failed! invalid json match operator")
```

Comment:
I would expect something like `"This trigger should have Failed! Invalid data"`)